### PR TITLE
Appointment dates

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/164396510

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt [hunk 1]
```diff
@@ -31,7 +31,13 @@ class ScheduleAppointmentSheetController @Inject constructor(
 
   private fun sheetCreates(events: Observable<UiEvent>): Observable<UiChange> {
     return events.ofType<ScheduleAppointmentSheetCreated>()
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt]: Done

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt [hunk 2]
```diff
@@ -31,7 +31,13 @@ class ScheduleAppointmentSheetController @Inject constructor(
...
-        .map { { ui: Ui -> ui.updateDisplayedDate(it.initialIndex) } }
+        .map {
+          { ui: Ui ->
+            ui.updateDisplayedDate(it.initialIndex)
+            ui.enableIncrementButton(it.initialIndex < it.size - 1)
+            ui.enableDecrementButton(it.initialIndex > 0)
+          }
+        }
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt]: done

File: app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEvents.kt [hunk 3]
```diff
@@ -3,7 +3,7 @@ package org.simple.clinic.scheduleappointment
 import org.simple.clinic.widgets.UiEvent
 import java.util.UUID
 
-data class ScheduleAppointmentSheetCreated(val initialIndex: Int, val patientUuid: UUID) : UiEvent
+data class ScheduleAppointmentSheetCreated(val initialIndex: Int, val patientUuid: UUID, val size: Int) : UiEvent
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEvents.kt]: done

**Comment by saket** [comment]
Hey rakshak, we'll need a couple of changes here:

1. The decrement and increment buttons don't have a disabled selector so they always remain blue even in disabled state. This should be a simple fix.

```xml
android:tint="@color/scheduleappointment_increment_decrement_date"
```

```xml
scheduleappointment_increment_decrement_date.xml
<?xml version="1.0" encoding="utf-8"?>
<selector xmlns:android="http://schemas.android.com/apk/res/android">
  <item android:color="@color/grey2" android:state_enabled="false" />
  <item android:color="@color/color_accent" />
</selector>
```

2. The lowest value is `not now` on zeplin but we're only going till 1 day. If we decide that we are not going to do something, then we should also update the zeplin spec. Please do it?

3. There's no cap on the dates as per zeplin, but we've capped the dates to 1 month at the most. Akshay says we should atleast go till `12 months`. Can you make this change and also update the spec on zeplin? Just keep in mind that having stale spec on zeplin is bad.

4. Because we're touching this sheet, let's update the sheet's layout to match it with zeplin. The design was recently updated.

**Comment by rakshakhegde** [comment]
@saket All the comments have been addressed. Please review again.

**Pull request merged**